gpytorch/models/exact_gp.py

```diff
@@ -112,15 +112,13 @@ def get_fantasy_model(self, inputs, targets, **kwargs):
 
         model_batch_shape = self.train_inputs[0].shape[:-2]
 
-        if self.train_targets.dim() > len(model_batch_shape) + 1:
-            raise RuntimeError("Cannot yet add fantasy observations to multitask GPs, but this is coming soon!")
-
         if not isinstance(inputs, list):
             inputs = [inputs]
 
         inputs = [i.unsqueeze(-1) if i.ndimension() == 1 else i for i in inputs]
 
-        target_batch_shape = targets.shape[:-1]
+        data_dim_start = -1 if targets.dim() <= inputs[0].dim() - 1 else -2
```

Comment:
@Balandat I think that will fix the issues you are seeing.

1. Do you think we could move some of these tests that are failing upstream in to gpytorch?
2. Do you think there's any use case for "I have a batch of data and I want to make a fantasy model where each batch gets the same label?" I ask because this is the other possible semantic meaning of `targets.dim() < input.dim()`.

Reply:
1. Sure, especially the faster fast fantasies (single set of inputs, batch of labels) would be good to have upstream. One of the tests failing is [this one](https://github.com/pytorch/botorch/blob/master/test/models/test_gpytorch.py#L61-L86) - we can just subclass `SimpleGPyTorchModel` only from `ExactGP` and throw out the botorch specifics, the test will still fail on this commit. There are others that can probably be adjusted in a similar fashion. 

2. By "the same label" you mean each input batch has the same training targets? I don't think there is a use case for this, at least not on our end. And even if there were, one can just expand the targets to the appropriate size - the savings in the computation come from computing the covariance once and batch solving that - we wouldn't have that in the case you're describing.

Reply:
So the difficulty here is to distinguish between the meaning of the last dimension of the targets - whether this is the dimension of the observation or whether it is the task/output dimension.

E.g. if you pass in a `n x d` input and a `n x n` target, it's not clear whether the target is `n` points of an `n`-dimensional outcome, or `n` batches of `n` points of a single-dimensional outcome. One could look at `self.train_targets`, but there we run into the same problem. 

The easiest way to ensure consistency here is to require an explicit outcome dimension for the targets. 

cc @sdaulton who may also have some thoughts on this.

Reply:
Well, so `n x d` inputs and `n x n` targets I don't think would be a valid input to this method. If you wanted `n` batches, it would have to be something like `n x n x 1` inputs and `n x n` targets.

I suppose we could check the existing prior distribution in the prediction strategy for being a `MultitaskMultivariateNormal`, and if so get the number of tasks from that, which would imply what `data_dim_start` should be.

Reply:
> Well, so n x d inputs and n x n targets I don't think would be a valid input to this method. If you wanted n batches, it would have to be something like n x n x 1 inputs and n x n targets.

Would it not be? If I don't want batches, but just have `n` data points in `d` dimensions, and happen to have a model with `n` outputs, the input dimensions above seem as if they should be correct?  

> I suppose we could check the existing prior distribution in the prediction strategy for being a MultitaskMultivariateNormal, and if so get the number of tasks from that, which would imply what data_dim_start should be.

This seems like it should work.

Reply:
@Balandat I think this is resolved now.